src/app/shared/plnkr/plunker-writer.ts

```diff
@@ -0,0 +1,128 @@
+import { Http } from '@angular/http';
+import { PlunkerData } from './plunker-data';
+
+/**
+ * Plunker writer, write example files to Plunker
+ *
+ * Plunker API
+ * URL: http://plnkr.co/edit/?p=preview
+ * data: {
+ *   // File name, directory and content of files
+ *   files[file-name1]: file-content1,
+ *   files[directory-name/file-name2]: file-content2,
+ *   // Can add multiple tags
+ *   tags[0]: tag-0,
+ *   // Description of plunker
+ *   description: description,
+ *   // Private or not
+ *   private: true
+ * }
+ */
+export class PlunkerWriter {
```

Comment:
You can add the `@Injectable` decorator to this to turn it into a service and add it to the `providers` list of the app's `NgModule`. Then `PlunkerButton` can inject this (instead of http) and this service can inject `Http` directly.